File: core/src/main/java/at/stefangeyer/challonge/model/Tournament.java
package at.stefangeyer.challonge.model;

import at.stefangeyer.challonge.model.enumeration.RankedBy;
import at.stefangeyer.challonge.model.enumeration.TieBreak;
import at.stefangeyer.challonge.model.enumeration.TournamentState;
import at.stefangeyer.challonge.model.enumeration.TournamentType;
import at.stefangeyer.challonge.model.query.enumeration.GrandFinalsModifier;
import lombok.Builder;
import lombok.Data;
import lombok.EqualsAndHashCode;

import java.time.OffsetDateTime;
import java.util.List;

@Data
@Builder
@EqualsAndHashCode(exclude = {
        "startAt", "completedAt", "createdAt",
        "predictionsOpenedAt", "startedAt", "startedCheckingInAt",
        "updatedAt", "lockedAt"})
public class Tournament {
    private Long id;
    private String url;
    private String name;
    private TournamentType tournamentType;
    private String subdomain;
    private String description;
    private Boolean openSignup;
    private Boolean holdThirdPlaceMatch;
    private Boolean ranked;
    private Boolean predictTheLosersBracket;
    private Float pointsForMatchWin;
    private Float pointsForMatchTie;
    private Float pointsForGameWin;
    private Float pointsForGameTie;
    private Float pointsForBye;
    private Integer swissRounds;
    private RankedBy rankedBy;
    private Integer roundRobinIterations;
    private Float roundRobinPointsForGameWin;
    private Float roundRobinPointsForGameTie;
    private Float roundRobinPointsForMatchWin;
    private Float roundRobinPointsForMatchTie;
    private Boolean acceptAttachments;
    private Boolean hideForum;
    private Boolean showRounds;
    private Boolean privateOnly;
    private Boolean notifyUsersWhenTheTournamentEnds;
    private Boolean sequentialPairings;
    private Float registrationFee;
    private String registrationType;
    private Integer signupCap;
    private OffsetDateTime startAt;
    private Long checkInDuration;
    private Boolean allowParticipantMatchReporting;
    private Boolean anonymousVoting;
    private String category;
    private OffsetDateTime completedAt;
    private OffsetDateTime createdAt;
    private Boolean createdByApi;
    private Boolean creditCapped;
    private Long gameId;
    private Boolean groupStagesEnabled;
    private Boolean hideSeeds;
    private Integer maxPredictionsPerUser;
    private Boolean notifyUsersWhenMatchesOpen;
    private Integer participantsCount;
    private Integer predictionMethod;
    private OffsetDateTime predictionsOpenedAt;
    private Integer progressMeter;
    private Boolean quickAdvance;
    private Boolean requireScoreAgreement;
    private OffsetDateTime startedAt;
    private OffsetDateTime startedCheckingInAt;
    private TournamentState state;
    private Boolean teams;
    private List<TieBreak> tieBreaks;
    private OffsetDateTime updatedAt;
    private String descriptionSource;
    private String fullChallongeUrl;
    private String liveImageUrl;
    private String signUpUrl;
    private Boolean reviewBeforeFinalizing;
    private Boolean acceptingPredictions;
    private Boolean participantsLocked;
    private String gameName;
    private Boolean participantsSwappable;
    private Boolean teamConvertable;
    private Boolean groupStagesWereStarted;
    private OffsetDateTime lockedAt;
    private Long eventId;
    private Boolean publicPredictionsBeforeStartTime;
    private GrandFinalsModifier grandFinalsModifier;
    private List<Participant> participants;
    private List<Match> matches;
}

File: src/main/kotlin/at/stefangeyer/challonge/model/Tournament.kt
/*
 * Copyright 2017 Stefan Geyer <stefangeyer at outlook.com>
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package at.stefangeyer.challonge.model

import at.stefangeyer.challonge.model.enum.RankedBy
import at.stefangeyer.challonge.model.enum.TournamentState
import at.stefangeyer.challonge.model.enum.TournamentType
import at.stefangeyer.challonge.model.query.enum.GrandFinalsModifier
import java.time.OffsetDateTime

class Tournament(
        var id: Long = 0L,
        var url: String = "",
        var name: String? = null,
        var tournamentType: TournamentType = TournamentType.SINGLE_ELIMINATION,
        var subdomain: String? = null,
        var description: String? = null,
        var openSignup: Boolean = false,
        var holdThirdPlaceMatch: Boolean = false,
        var pointsForMatchWin: Float = 0.0F,
        var pointsForMatchTie: Float = 0.0F,
        var pointsForGameWin: Float = 0.0F,
        var pointsForGameTie: Float = 0.0F,
        var pointsForBye: Float = 0.0F,
        var swissRounds: Int = 0,
        var rankedBy: RankedBy? = null,
        var roundRobinPointsForGameWin: Float = 0.0F,
        var roundRobinPointsForGameTie: Float = 0.0F,
        var roundRobinPointsForMatchWin: Float = 0.0F,
        var roundRobinPointsForMatchTie: Float = 0.0F,
        var acceptAttachments: Boolean = false,
        var hideForum: Boolean = false,
        var showRounds: Boolean = false,
        var private: Boolean = false,
        var notifyUsersWhenTheTournamentEnds: Boolean = false,
        var sequentialPairings: Boolean = false,
        var signupCap: Int? = 0,
        var startAt: OffsetDateTime? = null,
        var checkInDuration: Long? = 0,
        var allowParticipantMatchReporting: Boolean = false,
        var anonymousVoting: Boolean = false,
        var category: String? = null,
        var completedAt: OffsetDateTime? = null,
        var createdAt: OffsetDateTime? = null,
        var createdByApi: Boolean = false,
        var creditCapped: Boolean = false,
        var gameId: Long? = null,
        var groupStagesEnabled: Boolean = false,
        var hideSeeds: Boolean = false,
        var maxPredictionsPerUser: Int = 0,
        var notifyUsersWhenMatchesOpen: Boolean = false,
        var participantsCount: Int = 0,
        var predictionMethod: Int = 0,
        var predictionsOpenedAt: OffsetDateTime? = null,
        var progressMeter: Int = 0,
        var quickAdvance: Boolean = false,
        var requireScoreAgreement: Boolean = false,
        var startedAt: OffsetDateTime? = null,
        var startedCheckingInAt: OffsetDateTime? = null,
        var state: TournamentState = TournamentState.PENDING,
        var teams: Boolean? = false,
        var tieBreaks: List<String>? = listOf(),
        var updatedAt: OffsetDateTime? = null,
        var descriptionSource: String? = null,
        var fullChallongeUrl: String? = null,
        var liveImageUrl: String? = null,
        var signUpUrl: String? = null,
        var reviewBeforeFinalizing: Boolean = false,
        var acceptingPredictions: Boolean = false,
        var participantsLocked: Boolean = false,
        var gameName: String? = null,
        var participantsSwappable: Boolean = false,
        var teamConvertable: Boolean = false,
        var groupStagesWereStarted: Boolean = false,
        var lockedAt: OffsetDateTime? = null,
        var eventId: Long? = 0L,
        var publicPredictionsBeforeStartTime: Boolean? = false,
        var grandFinalsModifier: GrandFinalsModifier? = null,
        var participants: List<Participant> = listOf(),
        var matches: List<Match> = listOf()
)